reader-osm/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -219,14 +217,14 @@ private void prepareRestrictionRelation(ReaderRelation relation) {
     /**
      * @return all required osmWayIds to process e.g. relations.
      */
-    private TLongSet getOsmWayIdSet() {
+    private LongSet getOsmWayIdSet() {
         return osmWayIdSet;
     }
 
-    private TIntLongMap getEdgeIdToOsmWayIdMap() {
+    private IntLongMap getEdgeIdToOsmWayIdMap() {
         if (edgeIdToOsmWayIdMap == null)
-            edgeIdToOsmWayIdMap = new TIntLongHashMap(getOsmWayIdSet().size(), 0.5f, -1, -1);
-
+            edgeIdToOsmWayIdMap = new GHIntLongHashMap(getOsmWayIdSet().size(), 0.5f);
```

Comment:
Here we do not return -1 anymore if empty but we can easily check if the key is contained via the direct method